catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -796,22 +796,29 @@ QuerySources initializeSources(
           for (String id : sourceIds) {
             LOGGER.debug("Looking up source ID = {}", id);
 
-            final List<FederatedSource> fedSources =
+            List<FederatedSource> sources = new ArrayList<>();
+            sources.addAll(
                 frameworkProperties
                     .getFederatedSources()
                     .stream()
                     .filter(e -> e.getId().equals(id))
-                    .collect(Collectors.toList());
+                    .collect(Collectors.toList()));
+            sources.addAll(
+                frameworkProperties
+                    .getCatalogStores()
```

Comment:
✏️ I'm not sure that it'll actually cause any functional issues, but it may be good to use a `Set` to avoid duplicates since sources can be a `CatalogStore` _and_ a `FederatedSource`.

Reply:
I think there's issues with both. If we use a Set, we lose the ability to determine if multiple sources actually do exist for a single id. On the other hand, if we go with a List, we run into the scenario where we log a debug message saying there's multiple sources for an id when it was really just one source that was counted twice.